tests/test_util.py

```diff
@@ -240,8 +241,18 @@ def test_B9_persist_temp_file(self):
     dest_path = os.path.join(dest_temp_dir, self.random_string())
     tmpfile = tempfile.TemporaryFile()
     tmpfile.write(self.random_string().encode('utf-8'))
-    securesystemslib.util.persist_temp_file(tmpfile, dest_path)
+
+    # Write a file with restricted permissions
+    securesystemslib.util.persist_temp_file(tmpfile, dest_path, restrict=True)
     self.assertTrue(dest_path)
+
+    # Need to set also the stat.S_IFREG bit to match the st_mode output
+    # stat.S_IFREG - Regular file
+    expected_mode = stat.S_IFREG | stat.S_IRUSR | stat.S_IWUSR
+    if os.name != 'posix':
+      # Windows only supports setting the read-only attribute.
```

Comment:
Is this comment accurate? Doesn't 666 mean rw-rw-rw?

Reply:
It's _really_ hard to locate and parse the docs around this. 

Python's [`os.open()`](https://docs.python.org/3/library/os.html#os.open) points to the Windows [docs for `_open()`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen?redirectedfrom=MSDN&view=msvc-170) in their C Runtime which say that possible modes on Windows are:


pmode | Meaning
-- | --
_S_IREAD | Only reading permitted.
_S_IWRITE | Writing permitted. (In effect, permits reading and writing.)
_S_IREAD \| _S_IWRITE | Reading and writing permitted.

which I read as being effectively 0o666 on Windows. Some additional confusion creeps in because Python's [`os.chmod()`](https://docs.python.org/3/library/os.html#os.chmod) docs state:

> Note Although Windows supports [chmod()](https://docs.python.org/3/library/os.html#os.chmod), you can only set the file’s read-only flag with it (via the stat.S_IWRITE and stat.S_IREAD constants or a corresponding integer value). All other bits are ignored.

but I assume this means the mode cannot be changed from `stat.S_IREAD` to `stat.S_IWRITE`. However, I don't have access to a Windows box to verify. Nor do we have CI set up for Windows, so perhaps that (having Windows testing) is a blocker for this PR?

Reply:
Thanks for the detailed research. Just ran the test on a windows box and it works. Having Windows CI does not seem to be a blocker for this PR.

Reply:
OK, great. I think I will push an extra commit to change the `os.name != 'posix'` to `os.name == 'nt'` for additional clarity (I don't expect anyone is using sslib on QNX, but 🤷).

Reply:
Added that change as a suggestion https://github.com/secure-systems-lab/securesystemslib/pull/231/files#r988760854